src/common/auth.rs

```diff
@@ -1,14 +1,18 @@
+use rbac::JwtParser;
+
 use super::strings::ct_eq;
 use crate::settings::ServiceConfig;
 
 /// The API keys used for auth
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 pub struct AuthKeys {
     /// A key allowing Read or Write operations
     read_write: Option<String>,
 
     /// A key allowing Read operations
     read_only: Option<String>,
+
+    jwt_parser: Option<JwtParser>,
 }
```

Comment:
I have a suggestion, but open to discussion:

We can keep `AuthKeys` as is, to encapsulate the raw keys (the data layer). But then also add a secondary structure `Auth` to hold the keys and the jwt parser, which can act as the validation layer. 

Something like:
```suggestion
#[derive(Clone)]
pub struct AuthKeys {
    /// A key allowing Read or Write operations
    read_write: Option<String>,

    /// A key allowing Read operations
    read_only: Option<String>,
}

pub struct Auth {
    api_keys: AuthKeys,
    jwt_parser: Option<JwtParser>,
}
```

Reply:
Do you want to keep `can_read`/`can_write` methods in the `AuthKeys`, or move to `Auth` as well?

Also, I think that common validation logic could be placed in this validation layer. Maybe we should extract it later, when both actix and tonic middlewares would be in the same branch?

Reply:
Yeah, I also think we can generalize it better later. Let's postpone this